StreamChatClient_v3/ChatClient.swift

```diff
@@ -151,80 +142,264 @@ public class Client<ExtraData: ExtraDataTypes> {
+    /// The token of the current user. If the current user is anonymous, the token is `nil`.
+    @Atomic private var currentToken: Token?
+    
     /// Creates a new instance of Stream Chat `Client`.
     ///
     /// - Parameters:
-    ///   - currentUser: The user instance representing the current user of the chat.
     ///   - config: The config object for the `Client`. See `ChatClientConfig` for all configuration options.
-    public convenience init(currentUser: UserModel<ExtraData.User>, config: ChatClientConfig) {
+    public convenience init(config: ChatClientConfig) {
         // All production workers
         let workerBuilders: [WorkerBuilder] = [
-            MessageSender.init,
-            ChannelEventsHandler<ExtraData>.init
+            MessageSender.init
         ]
         
-        self.init(currentUser: currentUser,
-                  config: config,
+        self.init(config: config,
                   workerBuilders: workerBuilders,
                   environment: .init())
     }
     
     /// Creates a new instance of Stream Chat `Client`.
     ///
     /// - Parameters:
-    ///   - currentUser: The user instance representing the current user of the chat.
     ///   - config: The config object for the `Client`.
     ///   - workerBuilders: An array of worker builders the `Client` instance will instantiate and run in the background
     ///   for the whole duration of its lifetime.
     ///   - environment: An object with all external dependencies the new `Client` instance should use.
-    init(currentUser: UserModel<ExtraData.User>,
-         config: ChatClientConfig,
+    init(config: ChatClientConfig,
          workerBuilders: [WorkerBuilder],
          environment: Environment) {
         self.config = config
-        self.currentUser = currentUser
         self.environment = environment
         
         backgroundWorkers = workerBuilders.map { builder in
             builder(self.persistentContainer, self.webSocketClient, self.apiClient)
         }
     }
+    
+    deinit {
+        connectionIdWaiters.forEach { $0(nil) }
+        connectionIdWaiters.removeAll()
+    }
+    
+    /// Sets a new anonymous current user of the ChatClient.
+    ///
+    /// Anonymous users have limited set of permissions. A typical use case for anonymous users are livestream channels,
+    /// where they are allowed to read the conversation.
+    ///
+    /// - Parameters:
+    ///   - completion: Called when the new anonymous user is set. If setting up the new user fails, the completion
+    /// is called with an error.
+    public func setAnonymousUser(shouldConnectAutomatically: Bool = true, completion: ((Error?) -> Void)? = nil) {
+        disconnect()
+        prepareEnvironmentForNewUser(userId: .anonymous, role: .anonymous, extraData: nil) { error in
+            guard error == nil else {
+                completion?(error)
+                return
+            }
+            
+            self.connect(completion: completion)
```

Comment:
`shouldConnectAutomatically` ignored?

Reply:
Haha, I decided to remove it and forgot to remove it from the signature :) Forget about `shouldConnectAutomatically`...

Reply:
fixed